app/services/measurements_creator.rb

```diff
@@ -2,6 +2,13 @@ class MeasurementsCreator
   SLICE_SIZE = 500
 
   def self.call(stream, measurements_attributes)
+    if stream.session.type == "FixedSession"
```

Comment:
I haven't studied the code paths much. Wouldn't it be enough to have this in `#call`? You wouldn't need the `if` there cause there's an early return there

Reply:
You mean `MeasurementCreator.new.call`? That's not possible - cause there is also `AsyncMeasurementsCreator` (it's called on line 9 in this file). And this async creator then calls directly `MeasurementCreator.new.call` so it's too late.

Reply:
Good point! We should check if the async is called only for mobile measurements. I have a gut feeling that the if could be related to the type of session.